Signal/src/ViewControllers/MessagesViewController.m

```diff
@@ -2210,35 +2210,32 @@ - (void)imagePickerController:(UIImagePickerController *)picker
 
 - (void)sendMessageAttachment:(SignalAttachment *)attachment
 {
+    OWSAssert([NSThread isMainThread]);
     // TODO: Should we assume non-nil or should we check for non-nil?
     OWSAssert(attachment != nil);
     OWSAssert(![attachment hasError]);
     OWSAssert([attachment mimeType].length > 0);
-    
-    DispatchMainThreadSafe(^{
```

Comment:
This method is now only called from `tryToSendAttachmentIfApproved:` which ensures that this will be called on the main thread.